pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1541,15 +1544,23 @@ def __init__(self, *args, **kwargs):
         NormalSpoke.__init__(self, *args, **kwargs)
         self.networking_changed = False
         self.network_control_box = NetworkControlBox(self.builder, nmclient, spoke=self)
-        self.network_control_box.hostname = self.data.network.hostname
-        self.network_control_box.current_hostname = network.current_hostname()
+        self._network_module = DBusCachedObserver(MODULE_NETWORK_NAME,
+                                                  MODULE_NETWORK_PATH,
+                                                  [MODULE_NETWORK_NAME])
+        self._network_module.connect()
+        self.network_control_box.hostname = self._network_module.proxy.Hostname
+        self.network_control_box.current_hostname = self._network_module.proxy.GetCurrentHostname()
+        self._network_module.proxy.CurrentHostnameChanged.connect(self._hostname_changed)
         self.network_control_box.connect("nm-state-changed",
                                          self.on_nm_state_changed)
         self.network_control_box.connect("device-state-changed",
                                          self.on_device_state_changed)
         self.network_control_box.connect("apply-hostname",
                                          self.on_apply_hostname)
 
+    def _hostname_changed(self, hostname):
+        gtk_call_once(self._update_hostname)
```

Comment:
I think you want to call `async_utils.run_in_loop()` method instead. Ideally as decorator.

Reply:
I'll change it globally when I get to the network GUI spoke